compiler/test-resources/type-printer/source-compatible

```diff
@@ -0,0 +1,4 @@
+scala> class Bag extends reflect.Selectable
+// defined class Bag
+scala> val m = new Bag { val f = 23; def g = 47; def h(i: Int): Int = i; var i = 101; type N = Int }
+val m: Bag{val f: Int; def g: Int; def h(i: Int): Int; def i: Int; def i_=(x$1: Int): Unit; type N = Int} = anon$1@403364e9
```

Comment:
This is the current value produced in this PR while the expected in the issue was:
```suggestion
val m: Bag{val f: Int; def g: Int; def h(i: Int): Int; var i: Int; def i_=(x$1: Int): Unit; type N = Int} = anon$1@403364e9
```
which I don't think we would be able to achieve with the information here. There is no way I can see that we can get the symbol for `i`, does anyone can help out?